File: fizzbuzz/fizzbuzz.rs
use std::io::Write;

/// semi-efficient implementation of fizzbuzz
fn main() {
    let mut s = String::with_capacity(10);
    for i in 1..101 {
        if i % 3 == 0 { s.push_str("Fizz"); }
        if i % 5 == 0 { s.push_str("Buzz"); }
        if s.is_empty() { s.push_str(&i.to_string()); }
        s.push_str("\n");
        std::io::stdout().write(s.as_bytes());
        s.clear();
    }
}

File: fizzbuzz/Fizzbuzz.cs
using System.IO;
using System.Text;

public class Fizzbuzz {
  public static void Main(string[] args) {
    StringBuilder s = new StringBuilder(10);
    for (int j = 0; j <100000; j++)
    for (int i = 1; i < 101; i++) {
      if (i % 3 == 0)
        s.Append("fizz");
      if (i % 5 == 0)
        s.Append("buzz");
      if (s.Length == 0)
        s.Append(i);
      System.Console.WriteLine(s);
      s.Clear();
    }
  }
}
